e2e/tests/test_smoke.py

```diff
@@ -0,0 +1,191 @@
+        next invocation of "make e2e" will re-create it.
+        '''
+
+        k8s_api = K8sApi()
+
+        # HACK
+        # 1. creating RBAC entites with a separate client fails
+        #    with "AttributeError: object has no attribute 'select_header_accept'"
+        # 2. utils.create_from_yaml cannot create multiple entites from a single file
+        subprocess.run(["kubectl", "create", "-f", "manifests/operator-service-account-rbac.yaml"])
+
+        for filename in ["configmap.yaml", "postgres-operator.yaml"]:
+            utils.create_from_yaml(k8s_api.k8s_client, "manifests/" + filename)
+
+        # submit the most recent operator image built on the Docker host
+        body = {
+            "spec": {
+                "template": {
+                    "spec": {
+                        "containers": [
+                            {
+                              "name": "postgres-operator",
+                              "image": os.environ['OPERATOR_IMAGE']
+                            }
+                        ]
+                    }
+                 }
+            }
+        }
+        k8s_api.apps_v1.patch_namespaced_deployment("postgres-operator", "default", body)
+
+        # TODO check if CRD is registered instead
+        Utils.wait_for_pod_start(k8s_api, 'name=postgres-operator', cls.RETRY_TIMEOUT_SEC)
+
+        # HACK around the lack of Python client for the acid.zalan.do resource
+        subprocess.run(["kubectl", "create", "-f", "manifests/minimal-postgres-manifest.yaml"])
+
+        Utils.wait_for_pod_start(k8s_api, 'spilo-role=master', cls.RETRY_TIMEOUT_SEC)
+
+    @timeout_decorator.timeout(TEST_TIMEOUT_SEC)
+    def test_master_is_unique(self):
+        """
+           Check that there is a single pod in the k8s cluster with the label "spilo-role=master".
+        """
+        k8s = K8sApi()
+        labels = 'spilo-role=master,version=acid-minimal-cluster'
+
+        num_of_master_pods = Utils.count_pods_with_label(k8s, labels)
+        self.assertEqual(num_of_master_pods, 1, "Expected 1 master pod, found {}".format(num_of_master_pods))
+
+    @timeout_decorator.timeout(TEST_TIMEOUT_SEC)
+    def test_scaling(self):
+        """
+           Scale up from 2 to 3 pods and back to 2 by updating the Postgres manifest at runtime.
+        """
+        k8s = K8sApi()
+        labels = "version=acid-minimal-cluster"
+
+        Utils.wait_for_pg_to_scale(k8s, 3, self.RETRY_TIMEOUT_SEC)
+        self.assertEqual(3, Utils.count_pods_with_label(k8s, labels))
+
+        # TODO `kind` pods may stuck in the Terminating state for a few minutes; reproduce/file a bug report
+        Utils.wait_for_pg_to_scale(k8s, 2, self.RETRY_TIMEOUT_SEC)
+        self.assertEqual(2, Utils.count_pods_with_label(k8s, labels))
+
+    @timeout_decorator.timeout(TEST_TIMEOUT_SEC)
+    def test_taint_based_eviction(self):
+        """
+           Add taint "postgres=:NoExecute" to node with master.
```

Comment:
please briefly comment on the test scenario, e.g., what is expected when a taint is added.It basically tests the master failover  ; if I remember correctly replicas will simply die with the node and later recreated by the stateful set

PS we actually have 3 nodes in this setup, not sure if the master node is schedulable though